### DISCUSSION TITLE
Adding authentication page before loading the device list

### DISCUSSION THREAD

**Discussion by arun925** [NetrisTV/ws-scrcpy#73] [General]
As it is less secured planning to add authentication page for it.

any leads where I can add the authentication part ? May be an ldap based one

**Comment by drauggres** [comment]
Hi.
You will need to rewrite `/src/server/services/HttpServer.ts` module.
Now it is simplest HTTP server. You might want to use [express](https://www.npmjs.com/package/express) and [passport-ldapauth](https://www.npmjs.com/package/passport-ldapauth).

And since we are talking about security, there are several things to consider:
* you should use HTTPS on the server
* you need also a secure connection between browser and android-device: easiest way to get this (after you done with auth and ssl) is to use only proxy connection (`proxy over adb` option) and tell WebSocket-server in scrcpy to listen only on localhost: replace [this line](https://github.com/NetrisTV/scrcpy/blob/531b190706a7f8188cb66478025c9985a15fc018/server/src/main/java/com/genymobile/scrcpy/WSServer.java#L29) with `super(new InetSocketAddress("127.0.0.1", portNumber));`

I haven't tested any of this, so let me know if it works.
